src/cascade/executor/argument_parser.py

```diff
@@ -157,17 +158,18 @@ def _logging_configure_root_log(code_log_dir, level):
                                 f"Not making a log file for code log.")
                 return
 
-        fname = datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
-        try:
-            log_file_descriptor, code_log_path = tempfile.mkstemp(
-                suffix=".log", prefix=fname, dir=user_code_dir, text=True)
-        except (OSError, PermissionError) as ose:
-            logging.warning(f"Could not write to file in {user_code_dir} to make a log. {ose}")
+        if not os.access(str(user_code_dir), os.W_OK):
+            logging.warning(f"Could not write to file in {user_code_dir} to make a log")
             return
 
-        os.close(log_file_descriptor)
+        # On collision, this will fail to write a code log.
+        # Not using tempfile because it insists on secure mode flags.
+        uniquish = ''.join(choice(string.ascii_letters) for _ in range(8))
```

Comment:
Why not the token_urlsafe function? That should also make filename safe sequences and is simpler